modules/concurrency/include/hephaestus/concurrency/any_sender.h

```diff
@@ -0,0 +1,291 @@
+//=================================================================================================
+// Copyright (C) 2023-2025 HEPHAESTUS Contributors
+//=================================================================================================
+
+#pragma once
+
+#include <exception>
+#include <memory>
+#include <type_traits>
+#include <utility>
+
+#include <exec/any_sender_of.hpp>
+#include <stdexec/__detail/__execution_fwd.hpp>
+#include <stdexec/__detail/__senders_core.hpp>
+#include <stdexec/execution.hpp>
+
+namespace heph::concurrency {
+namespace internal {
+using AnySchedulerCompletions =
+    stdexec::completion_signatures<stdexec::set_error_t(std::exception_ptr), stdexec::set_stopped_t()>;
+using AnyScheduler = exec::any_receiver_ref<AnySchedulerCompletions>::any_sender<>::any_scheduler<>;
+static_assert(stdexec::scheduler<AnyScheduler>);
+
+/// Type erasing implementation for a stdexec environment forwarding a \ref AnyScheduler for
+/// `stdexec::get_scheduler` and `stdexec::get_delegation_scheduler` as well as forwarding
+/// `stdexec::get_stop_token`
+class AnyEnv {
+#ifndef DOXYGEN
+  struct Base {
+    virtual ~Base() = default;
+
+    [[nodiscard]] virtual auto getScheduler() const -> AnyScheduler = 0;
+    [[nodiscard]] virtual auto getStopToken() const -> stdexec::inplace_stop_token = 0;
+  };
+
+  template <typename Env>
+  class Impl : public Base {
+  public:
+    template <typename EnvT>
+      requires(std::is_same_v<Env, std::decay_t<EnvT>>)
+    explicit Impl(EnvT&& env) : env_(std::forward<EnvT>(env)) {
+    }
+    [[nodiscard]] auto getScheduler() const -> AnyScheduler final {
+      return stdexec::query_or(stdexec::get_scheduler, env_, stdexec::inline_scheduler{});
+    }
+
+    /// Forwards the stop token of the passed environment. If `stdexec::get_stop_token`
+    /// does not return `stdexec::inplace_stop_token`, returns a default constructed one which
+    /// never stops.
+    [[nodiscard]] auto getStopToken() const -> stdexec::inplace_stop_token final {
+      auto stop_token = stdexec::get_stop_token(env_);
+      using StopTokenT = decltype(stop_token);
+      if constexpr (std::is_same_v<stdexec::inplace_stop_token, StopTokenT>) {
+        return stop_token;
+      } else {
+        // NOTE: This is a stop-gap measure to allow to always return a stop token even though it
+        // might never fire.
+        const static stdexec::inplace_stop_source stop_source;
+        return stop_source.get_token();
```

Comment:
Is this safe? Does the `inplace_stop_source` ever stop in this case?

If we never expect this to happen in real code, can we statically assert that `std::is_same_v<stdexec::inplace_stop_token, StopTokenT>`?